File: rb/features.rb
# HikingTrailClosures SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module HikingTrailClosuresFeatures
  def self.make_feature(name)
    case name
    when "base"
      HikingTrailClosuresBaseFeature.new
    when "ratelimit"
      HikingTrailClosuresRatelimitFeature.new
    when "retry"
      HikingTrailClosuresRetryFeature.new
    when "test"
      HikingTrailClosuresTestFeature.new
    when "timeout"
      HikingTrailClosuresTimeoutFeature.new
    else
      HikingTrailClosuresBaseFeature.new
    end
  end
end

File: py/features.py
# HikingTrailClosures SDK feature factory

from feature.base_feature import HikingTrailClosuresBaseFeature
from feature.test_feature import HikingTrailClosuresTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HikingTrailClosuresBaseFeature(),
        "test": lambda: HikingTrailClosuresTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
